### ISSUE TITLE
No binaries found

### ISSUE THREAD

**Issue by wesleycoder** [vanaware/fnm-alpine#1]
Sorry for bothering, I know this isn't the official fnm repo.
I came here from Schniz/fnm/issues/694 where [this comment](https://github.com/Schniz/fnm/issues/694#issuecomment-1528881195) points to this repo.

I have this on my `.zshrc`:
```sh
# .zshrc
export FNM_HOME="$HOME/.local/share/fnm"
if [ ! -x $FNM_HOME ] >/dev/null; then
  curl -fsSL https://vanaware.github.io/fnm-alpine/install.sh | zsh -s -- --install-dir $FNM_HOME --skip-shell
fi

export PATH="$FNM_HOME:$PATH"
eval "`fnm env --use-on-cd --shell zsh --log-level=info`"
eval "`fnm completions --shell zsh`"
```

This is my shell output when trying to install.
```sh
$ fnm install 18.17.1
Installing Node v18.17.1 (x64)
error: Can't download the requested binary: v18.17.1 for x64 not found upstream.
You can `fnm ls-remote` to see available versions or try a different `--arch`.
```

Is there anything I may have set up wrong?
I'm trying to run fnm on alpine through wsl on windows.
It's a very sad use-case but I was hopeful to get some help if possible.
I can understand if this is out of your hands too.
Thanks in advance.

**Comment by Vinfall** [comment]
From https://github.com/Schniz/fnm/compare/master...vanaware:fnm-alpine:master, this fork _should_ work.

However, your output said `Installing Node v18.17.1 (x64)`, which means it's still requesting the `x64` build instead of `x64-musl`. (This is based on the assumption that https://unofficial-builds.nodejs.org/download/release/v18.17.1/ exists but it does so anyway...)

You can try my fork https://github.com/Vinfall/fnm/actions/runs/8642269712/artifacts/1404008438, it's really simple, just three lines of change to bypass the hardcoded arch check, see https://github.com/Schniz/fnm/commit/0562c9f5b35f81e4478fa3f25ec128e4ac604b65.

**Comment by wesleycoder** [comment]
Thanks for the follow up!
Also thanks for the good work!

I've switched my setup since I opened this issue, and nowadays this issue doesn't impact me anymore, so I'll be closing this issue for now.

I'll have some spare time this weekend to test the mentioned solution.
If I find any problems I'll come back here and reopen this issue or open a more appropriate one with better description if needed.

If everything goes well I may drop by here and leave a praise :)

**Comment by Vinfall** [comment]
fnm is just a binary (instead of shell alias like nvm) so the test should be easy, just download the artifact, unzip and mv it to `$FNM_DIR/fnm`.

You should also set shell rc  as mentioned [here](https://github.com/Schniz/fnm/issues/694#issuecomment-2049004644) (since they are no longer hardcoded like this repo):

```sh
export FNM_NODE_DIST_MIRROR=https://unofficial-builds.nodejs.org/download/release
export FNM_ARCH=x64-musl
```

**Comment by Vinfall** [comment]
FYI, the original repo has released [v1.36.0](https://github.com/Schniz/fnm/releases/tag/v1.36.0) without the hardcoded arch check. You can use that instead.
